lambda/src/main/scala/pricemigrationengine/handlers/AmendmentHandler.scala

```diff
@@ -248,74 +250,85 @@ object AmendmentHandler extends CohortHandler {
+                estimatedNewPrice = estimatedNewPrice,
+                priceCap = Newspaper2025P1Migration.priceCap,
+                invoiceList = invoicePreviewBeforeUpdate
+              )
             )
-          )
-        case HomeDelivery2025 =>
-          ZIO.fromEither(
-            HomeDelivery2025Migration.amendmentOrderPayload(
-              cohortItem = item,
-              orderDate = LocalDate.now(),
-              accountNumber = account.basicInfo.accountNumber,
-              subscriptionNumber = subscriptionBeforeUpdate.subscriptionNumber,
-              effectDate = startDate,
-              zuora_subscription = subscriptionBeforeUpdate,
-              oldPrice = oldPrice,
-              estimatedNewPrice = estimatedNewPrice,
-              priceCap = HomeDelivery2025Migration.priceCap,
-              invoiceList = invoicePreviewBeforeUpdate
+          case HomeDelivery2025 =>
+            ZIO.fromEither(
+              HomeDelivery2025Migration.amendmentOrderPayload(
+                cohortItem = item,
+                orderDate = LocalDate.now(),
+                accountNumber = account.basicInfo.accountNumber,
+                subscriptionNumber = subscriptionBeforeUpdate.subscriptionNumber,
+                effectDate = startDate,
+                zuora_subscription = subscriptionBeforeUpdate,
+                oldPrice = oldPrice,
+                estimatedNewPrice = estimatedNewPrice,
+                priceCap = HomeDelivery2025Migration.priceCap,
+                invoiceList = invoicePreviewBeforeUpdate
+              )
             )
-          )
-        case Newspaper2025P3 =>
-          ZIO.fromEither(
-            Newspaper2025P3Migration.amendmentOrderPayload(
-              cohortItem = item,
-              orderDate = LocalDate.now(),
-              accountNumber = account.basicInfo.accountNumber,
-              subscriptionNumber = subscriptionBeforeUpdate.subscriptionNumber,
-              effectDate = startDate,
-              zuora_subscription = subscriptionBeforeUpdate,
-              oldPrice = oldPrice,
-              estimatedNewPrice = estimatedNewPrice,
-              priceCap = Newspaper2025P3Migration.priceCap,
-              invoiceList = invoicePreviewBeforeUpdate
+          case Newspaper2025P3 =>
+            ZIO.fromEither(
+              Newspaper2025P3Migration.amendmentOrderPayload(
+                cohortItem = item,
+                orderDate = LocalDate.now(),
+                accountNumber = account.basicInfo.accountNumber,
+                subscriptionNumber = subscriptionBeforeUpdate.subscriptionNumber,
+                effectDate = startDate,
+                zuora_subscription = subscriptionBeforeUpdate,
+                oldPrice = oldPrice,
+                estimatedNewPrice = estimatedNewPrice,
+                priceCap = Newspaper2025P3Migration.priceCap,
+                invoiceList = invoicePreviewBeforeUpdate
+              )
             )
-          )
-      }
+        }
```

Comment:
I think I would extract this section as it looks like pure code that could be entirely wrapped in a `ZIO.fromEither`.  Also having an `order` value shadowing an outer `order` value might be confusing.

Reply:
Good idea! Will do